server/controllers/authController.js

```diff
@@ -1,10 +1,30 @@
-// authController.js
+/*
+----
+Core Feature(s): User Authentication, Registration, Session Management, Password Change
+Expected Input Type: (body)
+Expected Input: Depending on the function, input varies (e.g., login requires email and password, register requires user details including first name, last name, email, password, and optional interests)
+Expected Output Structure: JSON object with message, login status, user's first name, last name, and email
+Expected Errors: Various error responses including Bad Request, Unauthorized, Not Found, Server Error
+Purpose: This file contains functions related to user authentication, registration, session management, and password change in a web application. It handles user login, registration, logout, checking login status, and changing passwords securely using bcrypt for password hashing. Each function handles specific user-related tasks and interacts with the database to perform necessary operations.
+----
+*/
```

Comment:
Tops of files should have this, that for is only for exported api modules: 
`/*********************************************************************************
	
FileName:
FileVersion:
Core Feature(s):
Purpose: (1-3 sentences)

*********************************************************************************/
`